### PR TITLE
Layers opacity slider component

### PR DESCRIPTION
**Pull request by almostcake**
Flexberry-ddau-slider component for setting layer's opacity. Now it is enabled for all types excluding 'group'.
Default opacity could be set in maplayer's settings, if no value provided, it is set as 1.0 (100%)
It uses bootstrap slider wrapped into ui-ember-slider component and follows DDAU pattern just like flexberry-ddau-checkbox

### CODE REVIEW

File: addon/models/new-platform-flexberry-g-i-s-map-layer.js [hunk 1]
```diff
@@ -25,7 +25,12 @@ let Model = Projection.Model.extend(NewPlatformFlexberyGISMapLayerModelMixin, Le
     let stringToDeserialize = this.get('settings');
     if (!Ember.isBlank(stringToDeserialize)) {
       try {
-        return JSON.parse(stringToDeserialize);
+        let result = JSON.parse(stringToDeserialize);
+        if (!Ember.get(result, 'opacity')) {
+          Ember.set(result, 'opacity', 1);
+        }
```
Comment (kuzkok) [inline addon/models/new-platform-flexberry-g-i-s-map-layer.js]: You shouldn't set default values here, may be in layer settings or other place

Comment (SeregaMatin) [inline addon/models/new-platform-flexberry-g-i-s-map-layer.js]: I agree. 1 must be set into layer.settings.opacity as default value in "onMapLayerAdd" action handler and in base-layer's setOpacity method. This logic is unnecessary here.

File: addon/templates/components/flexberry-ddau-slider.hbs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{{!--
+  Binding through get-helper value=(get this "value") is a way to achieve one way binding.
+  It makes value immutable within component, what is major rule of "Data Down Actions Up (DDAU)" pattern:
+  component should display data, but not mutate it by its own.
+--}}
+{{ui-slider
```
Comment (SeregaMatin) [inline addon/templates/components/flexberry-ddau-slider.hbs]: I won't insist to change already used 3dparty library, but see [semantic-ui-range](https://github.com/tyleryasaka/semantic-ui-range) it looks more similar to other semantic modules.

File: addon/templates/components/flexberry-maplayer.hbs [hunk 3]
```diff
@@ -41,6 +43,12 @@
       <div style="clear: both;"></div>
     {{/if}}
 
+    {{#if _canChangeOpacity}}
+      <label class="{{flexberryClassNames.opacityLabel}}" title="{{t "components.flexberry-maplayer.opacity" }}">
```
Comment (SeregaMatin) [inline addon/templates/components/flexberry-maplayer.hbs]: To user it looks similar to buttons, click on it causes expand/collapse animation on related tree-node (because css-class flexberryClassNames.preventExpandCollapse isn't applied), so usability is strange.
It is better to display current opacity somewhere near the slider (above or under it, as a tooltip maybe), isn't it?

File: addon/components/flexberry-maplayer.js [hunk 4]
```diff
@@ -340,6 +373,15 @@ let FlexberryMaplayerComponent = Ember.Component.extend(
     visibility: false,
 
     /**
+      Layer's opacity.
+
+      @property visibility
+      @type Int
+      @default 100
+    */
+    opacity: 100,
```
Comment (SeregaMatin) [inline addon/components/flexberry-maplayer.js]: Why in the model and in slider component opacity range is 0.01..1, but here is 0..10? It is strange.

File: addon/components/base-layer.js [hunk 5]
```diff
@@ -98,6 +100,27 @@ export default Ember.Component.extend(
     }),
 
     /**
+      This layer's opacity
+      @property opacity
+      @type Float
```
Comment (SeregaMatin) [inline addon/components/base-layer.js]: As I know JS doesn't have such types as 'Float', 'Int', 'Double' and so on, it has only 'Number' type.

File: addon/components/flexberry-ddau-slider.js [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+      @type String[]
+      @default ['readonly:read-only']
+    */
+    classNameBindings: ['readonly:read-only'],
+
+    /**
+      Component's min value.
+
+      @property min
+      @type float
```
Comment (SeregaMatin) [inline addon/components/flexberry-ddau-slider.js]: As I know JS doesn't have such types as 'Float', 'Int', 'Double' and so on, it has only 'Number' type.
Alsos see same other properties downside.

File: addon/templates/components/flexberry-maplayer.hbs (hunk 3 above)
Comment (almostcake) [inline addon/templates/components/flexberry-maplayer.hbs]: It was Lena's task to make opacity value always visible for user, just like on Rosreestr's cadastral map. Use case of that is following: user can set opacity to 0 and then just forget about that, and then he will be shocked 'Why that layer is not displaying??', some kind of that.
preventExpandCollapse wasn't applied for next reason: when user see the value of opacity and wants to change it, he often automatically clicks on that label or whatever else (we think so).  Dropdown's arrow is not obvious button to do that.
Basically, it was considered to use semantic UI popup and do not place slider in accordion's content, but the situation was the same as 'accordion inside accordion', it didn't work.
Possibly, it is some kind of idea to make it just 'label', but with the same behavior as now.

File: addon/templates/components/flexberry-ddau-slider.hbs (hunk 2 above)
Comment (almostcake) [inline addon/templates/components/flexberry-ddau-slider.hbs]: It is possibly good idea and the way to make all the controls with the same framework, but there were so much problems with semantic's elements inside accordion and I believe it doesn't worth time to try

File: addon/templates/components/flexberry-maplayer.hbs (hunk 3 above)
Comment (SeregaMatin) [inline addon/templates/components/flexberry-maplayer.hbs]: Ok. Thanks for the detailed answer.

File: addon/templates/components/flexberry-ddau-slider.hbs (hunk 2 above)
Comment (SeregaMatin) [inline addon/templates/components/flexberry-ddau-slider.hbs]: Ok. I gave this link just for information.
